### PR TITLE
Milestone 2 : Queries and update animals table

### PR DESCRIPTION
**Pull request by Chrissiku**
### [Add a column species of type string to the animals' table](https://github.com/Chrissiku/Vet-clinic-database/blob/update-animals-table/schema.sql#L13-L18).
Add a column species

### [Insert New data](https://github.com/Chrissiku/Vet-clinic-database/blob/update-animals-table/data.sql#L17-L35)
Insert new data

### [Inside a transaction update the animals' table by setting the species column to `unspecified`. Verify that change was made.](https://github.com/Chrissiku/Vet-clinic-database/blob/update-animals-table/queries.sql#L46-L55)
transaction 1

###[Roll back the change and verify that the species columns went back to the state before the transaction](https://github.com/Chrissiku/Vet-clinic-database/blob/update-animals-table/queries.sql#L55-L62)
Roll Back Transaction 1

### [Inside a transaction as new transation](https://github.com/Chrissiku/Vet-clinic-database/blob/update-animals-table/queries.sql#L63-L81)

      - [x] Update the animals' table by setting the species column to `digimon` for all animals that have a name ending in `mon`.
      - [x] Update the animals' table by setting the species column to `pokemon` for all animals that don't have species 
      already set.
      - [x] Commit the transaction.
      - [x] Verify that change was made and persists after commit.

| Transaction | Commit 
| --| -- | 
| transaction | commit transaction |

### [New transaction](https://github.com/Chrissiku/Vet-clinic-database/blob/update-animals-table/queries.sql#L83-L94)
- [x] deletes all records in the animals' table
- [x] Roll back the transaction.
- [x] verify if all records in the animals' table still exist
transaction 3

### [Inside a transaction](https://github.com/Chrissiku/Vet-clinic-database/blob/update-animals-table/queries.sql#L95-L108)
- [x] Delete all animals born after `Jan 1st, 2022`.
- [x] Create a savepoint for the transaction.
- [x] Update all animals' weight to be their `weight multiplied by -1`.
- [x] Rollback to the savepoint
- [x] Update all animals' `weights that are negative` to be their `weight multiplied by -1`.
- [x] Commit transaction
transaction 4

### [Write queries to answer the given questions](https://github.com/Chrissiku/Vet-clinic-database/blob/update-animals-table/queries.sql#L111-L135)
- [x] How many animals are there?
- [x] How many animals have never tried to escape?
- [x] What is the average weight of animals?
- [x] Who escapes the most, neutered or not neutered animals?
- [x] What is the minimum and maximum weight of each type of animal?
- [x] What is the average number of escape attempts per animal type of those born between 1990 and 2000?
questions 1-5
questions 6

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -14,3 +14,22 @@ VALUES
...
+    animals(
+        id,
+        name,
+        date_of_birth,
+        escape_attempts,
+        neutered,
+        weight_kg
+    )
+VALUES
+    (5, 'Charmander', '08/02/2020', 0, FALSE, -11),
```
Comment (zieeco) [inline data.sql]: - Please ensure this is a floating number by adding a zero. e.g `-11.0`

File: data.sql [hunk 2]
```diff
@@ -14,3 +14,22 @@ VALUES
...
+    (6, 'Plantmon', '15/11/2021', 2, TRUE, -5.7),
+    (7, 'Squirtle', '02/04/1993', 3, FALSE, -12.13),
+    (8, 'Angemon', '12/06/2005', 1, TRUE, -45),
```
Comment (zieeco) [inline data.sql]: - Please ensure this is a floating number by adding a zero. e.g `-45.0`

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @<STUDENT NAME>,


Great job so far! 👏 
You've done well, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- No linter errors.✔️
- Nice design. ✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments under the review._





## Optional suggestions
- N/A

**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: data.sql (hunk 2 above)
Comment (Chrissiku) [reply data.sql]: Thank you @zieeco for this constructive feedback

**Review by omarsalem7 — APPROVED** [review]
Hi @Chrissiku 😊, 
# STATUS: APPROVED! 🥇 💯 

giphy

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 🚀 
Congratulations! 🎉 

## Highlights
✅ Project has met all the technical aspects you needed for this project. Keep rocking! 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Chrissiku** [comment]
Thank you @omarsalem7 for the project apporoval

**Pull request merged**